### PR TITLE
Add support for gjs/gts files

### PR DESCRIPTION
**Pull request by vstefanovic97**
How this works:

* Use [content-tag](https://github.com/embroider-build/content-tag) to transpile `.gjs/.gts` into spec compliant Javascrip/Typescript see https://rfcs.emberjs.com/id/0931-template-compiler-api/
* Run `analyzeJSFile` on the transpiled JS
* Use Babel to extract templates string from transpiled JS and then run `analyzeHbsFile` on the templates
* Merge and return all the keys found

### CODE REVIEW

**Comment by drbzyx** [comment]
@Mikek2252 Hey, would you be able to review this PR? It would be great to have support for the new Template Tag components.

**Comment by Mikek2252** [comment]
@vstefanovic97 Looks mostly fine, but i believe `globby` returns a promise so the functions that return globby values should be awaited.

**Comment by vstefanovic97** [comment]
@Mikek2252 they are all awaited, only thing I did was remove extraneous `async` on function that just return a promise but the don't have any `await`'s inside them.
The places where I removed `await` from function invocations are where functions don't return a promise at all

**Comment by Mikek2252** [comment]
@vstefanovic97 My mistake, this looks fine to me, i'll merge this now and get a release out in the next few days

**Comment by vstefanovic97** [comment]
Thank you @Mikek2252!

**Comment by Turbo87** [comment]
> only thing I did was remove extraneous async on function that just return a promise but the don't have any await's inside them.

this leads to errors thrown from these functions to not be converted to promise rejections though, so this needs to be done **verry** carefully. it usually better to keep them as async functions and use `return await ...` inside.

**Comment by vstefanovic97** [comment]
@Turbo87 I think that is probably only a problem if you are using `then/catch` syntax and the function ends up throwing an error and it doesn't get handled by `catch`.

But if you are using `try/catch` with `async/await` to handle error/promise rejection then it should be fine since still we will catch the error.

Either way I don't see a difference where this can affect the current logic, but if you prefer I can add back `async` keyword to the function to make sure they always return a promise

**Comment by Turbo87** [comment]
@vstefanovic97 it is a problem because it breaks the assumption about Promise-returning functions. aside from that, the `async` keyword is also a way to communicate to other developers that this function returns a Promise. without the keywords you have to dive into the internals of the function to figure out the return types. 

tl;dr removing the `async` keyword can cause many subtle and often invisible issues

**Comment by Turbo87** [comment]
and if you don't believe me, take it from the W3C: https://www.w3.org/2001/tag/doc/promises-guide#always-return-promises 😉

**Comment by vstefanovic97** [comment]
@Turbo87 I see your point, I added back the `async` keyword, do you mind running the workflow again?

File: index.js [hunk 1]
```diff
@@ -177,11 +184,11 @@ function joinPaths(inputPathOrPaths, outputPaths) {
   }
 }
 
-async function analyzeFiles(cwd, files, options) {
+function analyzeFiles(cwd, files, options) {
   let allTranslationKeys = new Map();
 
   for (let file of files) {
-    let translationKeys = await analyzeFile(cwd, file, options);
+    let translationKeys = analyzeFile(cwd, file, options);
```
Comment (Turbo87) [inline index.js:191]: looks like there are still plenty of places where async/await was removed

Comment (vstefanovic97) [reply index.js:191]: @Turbo87 those don't actually return promises, they are sync functions.
So basically for some reason sync functions had `async` keyword which I would say is also confusing

**Comment by drbzyx** [comment]
Hey, sorry to be a pain but are there any updates on this PR?

**Comment by Mikek2252** [comment]
Apologies, @Turbo87 are you happy with the updates? if so i'll get this merged and create a release

**Comment by vstefanovic97** [comment]
Pinging @Turbo87 again, are we ok with the changes now?

**Comment by amk221** [comment]
Would love to get a release with this :yellow_heart:

**Comment by Mikek2252** [comment]
@vstefanovic97 Thank you for the contribution, version 4.5.0 has been released now
